Source/Controllers/Preferences/Panes/SPNetworkPreferencePane.m

```diff
@@ -230,73 +306,252 @@ - (void)updateSSHConfigPopUp
+                            [staleBookmarksString appendFormat:@"%@\n", staleFile.lastPathComponent];
+                            SPLog(@"fileNames adding stale file: %@", staleFile.lastPathComponent);
+                        }
+
+                        [staleBookmarksString setString:[staleBookmarksString dropSuffixWithSuffix:@"\n"]];
+
+                        NSView *helpView = [self modifyAndReturnBookmarkHelpView];
+
+                        [NSAlert createAccessoryAlertWithTitle:NSLocalizedString(@"App Sandbox Issue", @"App Sandbox Issue") message:[NSString stringWithFormat:NSLocalizedString(@"You have missing secure bookmarks:\n\n%@\n\nWould you like to request access now?", @"Would you like to request access now?"), staleBookmarksString] accessoryView:helpView primaryButtonTitle:NSLocalizedString(@"Yes", @"Yes")
+                                          primaryButtonHandler:^{
+                            SPLog(@"request access now");
+                            [self->errorFileNames removeAllObjects];
+                            SPPreferenceController *prefCon = [((SPAppController *)[NSApp delegate]) preferenceController];
+                            [prefCon showWindow:nil];
+                            [prefCon displayPreferencePane:prefCon->fileItem];
+
+                        } cancelButtonHandler:^{
+                            SPLog(@"No not now");
+                        }];
+                    }
+                }
+            }
+        }
+    }
 }
 
-- (void)chooseSSHConfig
-{
-	// retrieve the file manager in order to fetch the current user's home
-	// directory
-	NSFileManager *fileManager = [NSFileManager defaultManager];
-	NSURL *homeDirectory = nil;
-	if ([fileManager respondsToSelector:@selector(homeDirectoryForCurrentUser)]) {
-		homeDirectory = [fileManager homeDirectoryForCurrentUser];
-	} else {
-		homeDirectory = [NSURL fileURLWithPath:NSHomeDirectory()];
-	}
+- (BOOL)checkSSHConfigFileForUserKnownHostsFile:(NSString*)configFile{
 
-	_currentFilePanel = [NSOpenPanel openPanel];
-    [_currentFilePanel setMessage:NSLocalizedString(@"Please choose your ssh config files(s)", "Please choose your ssh config files(s)")];
-	[_currentFilePanel setCanChooseFiles:YES];
-	[_currentFilePanel setCanChooseDirectories:NO];
-	[_currentFilePanel setAllowsMultipleSelection:YES];
-	[_currentFilePanel setAccessoryView:hiddenFileView];
-	[_currentFilePanel setResolvesAliases:NO];
-	[_currentFilePanel setDirectoryURL:[homeDirectory URLByAppendingPathComponent:@".ssh"]];
-	[self updateHiddenFiles];
-	
-	[prefs addObserver:self
-			forKeyPath:SPHiddenKeyFileVisibilityKey
-			   options:NSKeyValueObservingOptionNew
-			   context:NULL];
-	
-	[_currentFilePanel beginWithCompletionHandler:^(NSInteger returnCode)
-	 {
-		// only process data, when the user pressed ok
-		if (returnCode != NSModalResponseOK) {
-			return;
-		}
+    SPLog(@"checkSSHConfigFileForUserKnownHostsFile");
```

Comment:
Pretty darn cool - we'll need to do some serious testing with all this stuff though. Looks awesome

Reply:
agree ... please help with testing 

- @Sequel-Ace/all 
- @sjlehn
- @effata 
- @wedi
- @mfehr94

Reply:
@jamesstout @Jason-Morcos You release a beta for this?

Reply:
> @jamesstout @Jason-Morcos You release a beta for this?

I can't. Need @Kaspik or @Jason-Morcos

Reply:
Once merged we will for sure

Reply:
@jamesstout Looks like one conflict. When resolved, let's merge this and then I can push a beta indeed